src/java.base/share/classes/sun/security/pkcs12/MacData.java

```diff
@@ -72,28 +83,65 @@ class MacData {
         AlgorithmId digestAlgorithmId = AlgorithmId.parse(digestInfo[0]);
         this.digestAlgorithmName = digestAlgorithmId.getName();
         this.digestAlgorithmParams = digestAlgorithmId.getParameters();
+
         // Get the digest.
         this.digest = digestInfo[1].getOctetString();
 
-        // Get the salt.
-        this.macSalt = macData[1].getOctetString();
+        if (digestInfo[0].tag != DerValue.tag_Sequence) {
+            throw new IOException("algid parse error, not a sequence");
+        }
+        if (digestAlgorithmName.equals("PBMAC1")) {
+            PBEParameterSpec pbeSpec;
+
+            try {
+                pbeSpec =
+                        digestAlgorithmParams.getParameterSpec(
+                        PBEParameterSpec.class);
+            } catch (InvalidParameterSpecException ipse) {
+                throw new IOException(
+                        "Invalid PBE algorithm parameters");
+            }
+            iterations = pbeSpec.getIterationCount();
+            macSalt = pbeSpec.getSalt();
```

Comment:
Nit, use `this` for class fields (`this.iterations`, `this.macSalt`, etc) to keep method code consistent and helps distinguish from local variables.